aws/resource_aws_msk_cluster.go

```diff
@@ -105,52 +107,54 @@ func resourceAwsMskCluster() *schema.Resource {
 				},
 			},
 			"client_authentication": {
-				Type:     schema.TypeList,
-				Optional: true,
-				ForceNew: true,
-				MaxItems: 1,
+				Type:             schema.TypeList,
+				Required:         true,
```

Comment:
My logic behind making both the `client_authentication` and `unauthenticated` attributes mandatory is to force users to explicitly set the authentication options.

Whilst this field is optional in the API the behaviour when creating a cluster without this set is undefined. 
eg. If we create a cluster without this set the AWS Console UI shows the cluster has all authentication modes disabled and the `describeCluster` endpoint omits the `client_authentication` key completely in its response, despite this the behaviour observed is that it is possible to connect unauthenticated.

If the user were to make an update and include this block without setting any authentication options then the `updateSecurity` call fails saying at least one auth method should be set.

If were to default the `Unauthenticated` option to true there is a danger that a user could inadvertently disable unauthenticated access, likewise the inverse if they have configuration already for `tls` then defaulting this to true could unknowingly enable Unauthenticated access when they re-apply.

If there is a better way or idea I am open to it, we could potentially only send the fields if they are explicitly set, perhaps detecting changes on the individual attributes instead of the whole block?

Reply:
For backwards compatibility we will have to keep everything as `Optional`.